### PR TITLE
[Survey Accounts] Data Frameworkify the module

### PR DESCRIPTION
**Pull request by driusan**
This updates the survey accounts module to use the data framework. Functionality should remain identical to before, except that "Site" and "Project" permissions should now be enforced by the menu filter and users should not see rows from other project/site's sessions.

### CODE REVIEW

**Comment by driusan** [comment]
Blocked by either #6047 or #6011.

File: modules/survey_accounts/php/surveyaccountsprovisioner.class.inc [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+/**
```
Comment (johnsaigle) [inline modules/survey_accounts/php/surveyaccountsprovisioner.class.inc]: You can delete these file comments now if you want as long as we merge #6046 first.

**Comment by johnsaigle** [comment]
1) Survey_AccountsTestIntegrationTest::testSurveyAccountsSearchButton
Failed asserting that '0 rows displayed of 0. (Maximum rows per page: 20501001000500010000)' contains "1 rows".
/app/modules/survey_accounts/test/survey_accountsTest.php:304
/app/modules/survey_accounts/test/survey_accountsTest.php:273

**Comment by driusan** [comment]
@kongtiaowang is there a good way to get the UserID to insert into the user_psc_rel and user_project_rel tables so that this test will still work as before?

**Comment by driusan** [comment]
@kongtiaowang ping

**Comment by driusan** [comment]
@kongtiaowang Can you please answer the above question?

**Comment by kongtiaowang** [comment]
@driusan There are some minor issues. "missing void" Return type is not compatible.
1.  URL links
" PHP Fatal error:  Declaration of NDB_BVL_Instrument_No_Additional_Options::_addMetadataFields() must be compatible with NDB_BVL_Instrument::_addMetadataFields(): void in /var/www/loris/project/libraries/NDB_BVL_Instrument_No_Additional_Options.class.inc on line 0, referer: https://wangshen-dev.loris.ca/survey_accounts/addSurvey/"
2. add Survey
 "PHP Fatal error:  Declaration of NDB_BVL_Instrument_No_Additional_Options::addMonthYear($field, $label, $options = Array) must be compatible with NDB_BVL_Instrument::addMonthYear(string $field, string $label, array $options = Array): void in /var/www/loris/project/libraries/NDB_BVL_Instrument_No_Additional_Options.class.inc on line 0, referer: https://wangshen-dev.loris.ca/survey_accounts"
3. 
PHP Fatal error:  Class NDB_BVL_Instrument_No_Additional_Options contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (NDB_BVL_Instrument::getFullName, NDB_BVL_Instrument::getSubtestList) in /var/www/loris/project/libraries/NDB_BVL_Instrument_No_Additional_Options.class.inc on line 0

**Comment by driusan** [comment]
@kongtiaowang even if those are changed there will still be the issue of the rows returned being 0 because of the project filter being added. How do we fix that?

**Comment by driusan** [comment]
@kongtiaowang and where are those errors coming from? I don't see any NDB_BVL_Instrument_No_Additional_Options in LORIS.

**Comment by kongtiaowang** [comment]
@driusan I found the issues when I test it manually. It can't add a new survey, so the result of the filter test result should be one, not zero.

**Comment by kongtiaowang** [comment]
@driusan the testing data can't insert into the survey table. I will figure it out.

**Comment by kongtiaowang** [comment]
@driusan The Survey_Accounts uses useSiteFilter and useProjectFilter, so the projectId and centerId of testing data do not match useSiteFilter and useProjectFilter.
to fix:
Screen Shot 2020-02-28 at 3 02 57 PM

**Comment by kongtiaowang** [comment]
@driusan So sorry. Please forget it can't add a new survey issue. That is because of my instrument files aren't updated.

**Comment by driusan** [comment]
@lingma I don't understand what you mean. It's mock data, it needs to match the table. The column is called RegistrationCenter/ProjectID in the candidate table and just center/project in the session table.

**Comment by lingma** [comment]
> @lingma I don't understand what you mean. It's mock data, it needs to match the table. The column is called RegistrationCenter/ProjectID in the candidate table and just center/project in the session table.
select RegistrationCenterID as CenterID could be used as CenterID, so it is not technical problem, just what the real need

**Comment by driusan** [comment]
The center/project is added for the filtering. Since there is a session in the query, the actual center of the session is more accurate/specific than the place where the candidate originally registered (which shouldn't grant them access to all of the candidate's data just because it was first.)
